syntax = "proto3";

package deepface.api.proto;

import "common.proto";

message RepresentRequest {
  bytes image = 1;
  optional deepface.api.proto.Models model_name = 2;
  optional deepface.api.proto.Detectors detector_backend = 3;
  optional bool enforce_detection = 4;
  optional bool align = 5;
  optional bool anti_spoofing = 6;
  optional uint32 max_faces = 7;
}

message RepresentResponse {

    message Results {
        repeated double embedding = 1;
        double face_confidence = 2;
        deepface.api.proto.FacialArea facial_area = 3;
    }

    repeated Results results = 1;
}

service RepresentService {
  rpc Represent(RepresentRequest) returns (RepresentResponse);
}